### ISSUE TITLE
Resetting navigation state on v1.0.0-beta.22

### ISSUE THREAD

**Issue by jamsch** [react-navigation/react-navigation#3149]
### Current Behavior
Given the following code that worked on v1.0.0-beta.21:

#### Navigator structure

```javascript
const AuthNavigator = StackNavigator({
  OnboardingScreen: {},
  LoginScreen: {},
  SignUpScreen: {},
});

const UserNavigator = DrawerNavigator({
    ProfileStack: {
      screen: StackNavigator({
        ProfileScreen: {
          screen: ProfileScreen,
          navigationOptions: {
            accessibilityLabel: 'Profile Screen',
          },
        },
        LeaderboardScreen: {},
      }),
    },
}, {
    headerMode: 'none',
    drawerWidth: 300,
    drawerPosition: 'left',
    contentComponent: props => <DrawerContent {...props} />,
});

export default StackNavigator(
  {
    Authentication: { screen: AuthNavigator }, 
    User: { screen: UserNavigator }, // Logged in
  },
  {
    initialRouteName: 'Authentication',
    headerMode: 'none',
  },
);

```

#### Reducer (this previously did work on v1.0.0-beta.21)
```javascript
function navReducer(state = initialState, action) {
  switch (action.type) {
    // Settings are retrieved from secondary storage
    case REHYDRATE: {
      // Check if user has been logged in
      if (action.payload.auth.status === "loggedin") {
          // Go to Profile
          return AppNavigator.router.getStateForAction(
            NavigationActions.reset({
              index: 0,
              key: null,
              actions: [
                NavigationActions.navigate({
                  routeName: "User",
                  params: { title: `${action.payload.user.name}'s profile` }
                })
              ]
            })
          );
        }     
    }
  }
};
```

#### Result from using reset:
<img src="https://i.imgur.com/FIa0be5.png" alt="alt text" width="300">

Adding `NavigationActions.init()` does the following:
```javascript
            return AppNavigator.router.getStateForAction(NavigationActions.reset({
              index: 0,
              key: null,
              actions: [
                NavigationActions.init({ routeName: 'User' }),
                NavigationActions.navigate({
                  routeName: 'ProfileScreen',
                  params: { title: `${action.payload.user.name}'s profile` }
                }),
              ],
            }));
```
#### Result from using reset with `NavigationActions.init`:
<img src="https://i.imgur.com/Myw8X5j.png" alt="alt text" width="300">

### Expected Behavior
The reset action should allow users to configure the navigation state from the root. Instead this doesn't seem to appear functional.

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation | v1.0.0-beta.22
| react-native     | v0.51.0
| node             | v6.10.3
| npm or yarn      | npm@3.10.10

**Comment by kelset** [comment]
Uhm thanks for reporting.
This is probably related to one of these two PRs we merged in latest:
* https://github.com/react-community/react-navigation/pull/895
* https://github.com/react-community/react-navigation/pull/3094

But I'm not really sure why.

**Comment by jamsch** [comment]
For more clarification I updated the navigation structure. One of the root navigators is a drawer navigator and that's the one I'm trying to reset to which doesn't allow me.

**Comment by jamsch** [comment]
What seems to fix it is setting the drawer routes in your navigation options. It now seems to be required as long as you've set navigation options for the drawer (as per pull request #1803). This should really be defaulted though.

```
{
    drawerOpenRoute: 'DrawerOpen',
    drawerCloseRoute: 'DrawerClose',
    drawerToggleRoute: 'DrawerToggle'
    headerMode: 'none',
    drawerWidth: 300,
    drawerPosition: 'left',
    contentComponent: props => <DrawerContent {...props} />,
}

```

**Comment by Uthpala** [comment]
Boy better testing needs to be done before things go to a release. Spent about 3 hours figuring out what was wrong. Finally to land here.

**Comment by anatolinicolae** [comment]
Same here, spent more than 3 hours digging into the code and rolling back commits (blaming self) since this issue didn't allow further app execution and development...

**Comment by flynnch** [comment]
add routeName: payload.routeName  at line 42  NavigationActions.js  @jamsch

**Comment by aMarCruz** [comment]
Same here, lost 2 hours... beta 22 has undocummented breaking changes.
Too bad for a 8k stars library

**Comment by kelset** [comment]
We are doing our best to have the library is a good spot - sadly in this situation, something slipped through.

I've modified the Release Notes accordingly to @jamsch last comment (thanks for that btw).

**Comment by kelset** [comment]
Quick update, we have merged a PR to master that should fix this issue (#3152)

**Comment by kelset** [comment]
Closing since `beta.25` has the fix merged.

**Comment by bhatti-waqas** [comment]
Thanks alot @jamsch  You comment helped me alot. and saved my more time to be wasted.

**Comment by jsdario** [comment]
Just updated from `beta-11` to `1.5.7` and found the same message, also because of the drawer, but none of the proposed solutions are working.

**Comment by brentvatne** [comment]
hi there! if this is still a bug can you create a new issue that reproduces the bug on https://snack.expo.io? thanks!
